### PR TITLE
Feature/cxspa 3400 template strict

### PR DESCRIPTION
**Pull request by ChristophHi**

### CODE REVIEW

File: feature-libs/product-configurator/rulebased/components/group-menu/configurator-group-menu.component.ts [hunk 1]
```diff
@@ -271,6 +271,22 @@ export class ConfiguratorGroupMenuComponent {
...
 
+  //TODO(CXSPA-3392) get rid of this method in next major. Change signature of
+  //isConflictGroupType to allow undefined, and use this method instead
+  /**
+   * Verifies whether the current group is conflict one but allows for undefined input
+   *
+   * @param {Configurator.GroupType} groupType - Group type
+   * @return {boolean} - 'True' if the current group is conflict one, otherwise 'false'.
+   */
+  isConflictGroupTypeWUndefined(
```
Comment (steinsebastian) [inline feature-libs/product-configurator/rulebased/components/group-menu/configurator-group-menu.component.ts]: Why is this method called isConflictGroupType**W**Undefined and not isConflictGroupTypeUndefined?

**Review by steinsebastian — CHANGES_REQUESTED** [review]
See comment

File: feature-libs/product-configurator/rulebased/components/group-menu/configurator-group-menu.component.ts (hunk 1 above)
Comment (ChristophHi) [reply feature-libs/product-configurator/rulebased/components/group-menu/configurator-group-menu.component.ts]: The method should check if a group is a conflict group, allowing undefined input.
What about isConflictGroupTypeAllowingUndefined?

Comment (steinsebastian) [reply feature-libs/product-configurator/rulebased/components/group-menu/configurator-group-menu.component.ts]: Sounds good.

**Review by steinsebastian — APPROVED** [review]

**Pull request merged**